### PR TITLE
WIP: Convert unsafe.CStringSuite.scala to JUnit unsafe.CStringTest.scala

### PR DESCRIPTION
**Pull request by LeeTibbert**
* PR #1945 added two cases to unsafe.CStringSuite.scala. To aid
    review of #1945, I did not convert CStringSuite from the
    classical Scala Native test format to JUnit. Rather, I left
    an IOU in the commit message for #1945.  This PR pays off
    that IOU.

  * There should be no functional changes. A trial testOnly of
    CSuiteTest.scala reveals no changes beyond changed test names
    and JUnit style output.

  * It looks as though the CStringSuite tests had been written at
    different times, with slightly different assert styles. I am
    supposing that the later authors did not want to alter existing,
    working tests, again to minimize and isolate changes.

    I changed some of the asserts to use a consistent "expected, result"
    format. This is the convention documented by JUnit and some/most of the
    existing tests.  Future readers will expect that a JUnit test
    follow the JUnit convention.

Documentation:

  * As a number of tests are changing to JUnit the Release Notes will
    need a blanket statement saying "Many tests changed to JUnit.".
    This test would be covered by that statement.

Testing:

  Safety -

  + Built and tested ("test-all") in debug mode using sbt 1.3.13 on
    X86_64 only . All tests pass.

  Efficacy -

  + JUnit style output is evident upon execution.

### CODE REVIEW

File: unit-tests/src/test/scala/scala/scalanative/unsafe/CStringTest.scala [hunk 1]
```diff
@@ -121,13 +120,13 @@ object CStringSuite extends tests.Suite {
...
       val jstr2: String = fromCString(arr.at(0))
 
-      assertEquals(strcmp(cstr1, cstr2), 0)
-      assertEquals(strlen(arr.at(0)), 10)
+      assertEquals(0, strcmp(cstr1, cstr2))
+      assertEquals(10, strlen(arr.at(0)))
 
       assertEquals(jstr2, jstr1 * 2)
       assertEquals(jstr2.last, 'd')
-      assertEquals(!(cstr1 + 4), 'd')
```
Comment (ekrich) [inline unit-tests/src/test/scala/scala/scalanative/unsafe/CStringTest.scala:129]: I believe these two tests were using pointer arithmetic and now they are using the index method.

Comment (LeeTibbert) [reply unit-tests/src/test/scala/scala/scalanative/unsafe/CStringTest.scala:129]: Thank you for the review.  My brain must be stuck in C land, where the two are exactly equivalent.  
By my understanding of Scala, they two ought to be equivalent in results (I am avoiding the word 'functionally'),
A CString is an Array[Byte]. One ought to be able to index off it.

I can restore the pointer math, if you think there is even the slightest chance that I introduced an error.
I thought that an Array index would be more natural to a Scala reviewer.  

Please advise.  My goal is to make people's work easier, not harder.

L

Comment (ekrich) [reply unit-tests/src/test/scala/scala/scalanative/unsafe/CStringTest.scala:129]: You could leave the test as is and add 2 new cases? When I was porting I didn't change anything for the most part. `assert -> assertTrue` for example. You changed to `assertEquals` which I think is okay but it widens the diff.

Comment (LeeTibbert) [reply unit-tests/src/test/scala/scala/scalanative/unsafe/CStringTest.scala:129]: IIUC, you are suggesting adding two cases using pointer math?  I can do that.

Based on your "widening" concerns and the cadence you and Sébastien have established
I am going to convert this to WIP and do the AssertEquals in a subsequent PR, once the
dust has settled.

Comment (ekrich) [reply unit-tests/src/test/scala/scala/scalanative/unsafe/CStringTest.scala:129]: I think the widening may be fine for one test but I would leave the pointer arithmetic test as is or add 2 new cases. I am not an expert but `assertEquals` is probably a big improvement.

**Comment by LeeTibbert** [comment]
Closing in favor of @ekrich PR #1989

**Pull request closed without merging**